runtime/src/accounts_db.rs

```diff
@@ -4231,7 +4231,15 @@ impl AccountsDb {
         let chunks = crate::accounts_hash::MERKLE_FANOUT.pow(4);
         let total_lamports = Mutex::<u64>::new(0);
         let get_hashes = || {
-            keys.par_chunks(chunks)
+            let mut hashes = vec![];
+            for ix in 0..self.accounts_index.account_maps.num_buckets() {
+                let keys = self.accounts_index.account_maps.keys(ix);
```

Comment:
@ryoqun @sakridge @jeffwashington 

wdyt, would this work here?  the initial design had a snapshot slot that would keep the data behind that slot form being cleaned up as the index is updated.  So we shouldn't need a global lock on all the keys to do the snapshot.

Reply:
@brooksprumo is good to reference here, as he's been working on snapshotting related logic

Reply:
I think this code path is already dead except for debug validation. We don’t want to use account index for hash calculation. Even so, does bucket[x] have keys that are all < the keys in bucket[x+1]? This algorithm is written to assume the keys vector is in sorted order. If not, we have to save key and hash from this scan, sort by key, then hash.

Reply:
I can confirm what @jeffwashington said that on the snapshot side this is never called/used. The snapshot code that calls `AccountsDb::calculate_accounts_hash_without_index()` is inside an `if` that is always false.

There are hooks for testing, but none of the tests actually exercise this code path. I assume in the past this was different, but as of now `AccountsDb::calculate_accounts_hash_without_index()` is never called in the snapshot code (nor anything that calls these snapshot functions).

Relevant snapshot functions:
`snapshot_utils::snapshot_bank()`
`snapshot_utils::package_snapshot()`

Reply:
I can confirm what @jeffwashington said that on the snapshot side this is never called/used. The snapshot code that calls `AccountsDb::calculate_accounts_hash_without_index()` is inside an `if` that is always false.

There are hooks for testing, but none of the tests actually exercise this code path. I assume in the past this was different, but as of now `AccountsDb::calculate_accounts_hash_without_index()` is never called in the snapshot code (nor anything that calls these snapshot functions).

Relevant snapshot functions:
`snapshot_utils::snapshot_bank()`
`snapshot_utils::package_snapshot()`

Reply:
@jeffwashington 
> Even so, does bucket[x] have keys that are all < the keys in bucket[x+1]? 

Yep.  that is guaranteed. https://github.com/solana-labs/solana/pull/18179/files#diff-d90ce8cbcea9459dbca60678b48f6d6d2816b8d4de8b8ebfd5c434b819f153b3R97